### PR TITLE
EPMRPP-47707: Allow plugins to add own tabs to project settings

### PR DESCRIPTION
**Pull request by DiscoElevator**
Waiting for #2322

### CODE REVIEW

File: app/src/routes/routesMap.js [hunk 1]
```diff
@@ -123,7 +113,7 @@ const routesMap = {
     },
   },
   [PROJECT_DETAILS_PAGE]: {
-    path: `/administrate/projects/:projectId/:projectSection(${SETTINGS}|${MEMBERS}|${EVENTS})?/:settingsTab(${GENERAL}|${NOTIFICATIONS}|${INTEGRATIONS}|${DEFECT}|${ANALYSIS}|${PATTERN_ANALYSIS}|${DEMO_DATA})?`,
+    path: `/administrate/projects/:projectId/:projectSection(${SETTINGS}|${MEMBERS}|${EVENTS})?/:settingsTab?`,
```
Comment (AmsterGet) [inline app/src/routes/routesMap.js]: So how will unknown tabs be handled? (see the screenshot below for version 5.1)

image

File: app/src/routes/routesMap.js [hunk 2]
```diff
@@ -254,7 +244,7 @@ const routesMap = {
     type: PROJECT_SETTINGS_TAB_PAGE,
     payload: { ...payload, settingsTab: GENERAL },
   })),
-  PROJECT_SETTINGS_TAB_PAGE: `/:projectId/settings/:settingsTab(${GENERAL}|${NOTIFICATIONS}|${INTEGRATIONS}|${DEFECT}|${ANALYSIS}|${PATTERN_ANALYSIS}|${DEMO_DATA})`,
+  PROJECT_SETTINGS_TAB_PAGE: `/:projectId/settings/:settingsTab`,
```
Comment (AmsterGet) [inline app/src/routes/routesMap.js]: The same as above

File: app/src/routes/routesMap.js (hunk 1 above)
Comment (DiscoElevator) [reply app/src/routes/routesMap.js]: I can show the first tab of the list. Is it ok for now?

Comment (DiscoElevator) [reply app/src/routes/routesMap.js]: (Now it throws error because no config for this unknow tab found)

Comment (AmsterGet) [reply app/src/routes/routesMap.js]: > I can show the first tab of the list. Is it ok for now?

Yes, it is.

File: app/src/components/main/navigationTabs/navigationTabs.jsx [hunk 3]
```diff
@@ -51,6 +51,25 @@ export class NavigationTabs extends Component {
     customBlock: null,
   };
 
+  componentDidMount() {
+    const { activeTab, config } = this.props;
```
Comment (AmsterGet) [inline app/src/components/main/navigationTabs/navigationTabs.jsx]: I think will be good to move this check and action into separate function

Comment (DiscoElevator) [reply app/src/components/main/navigationTabs/navigationTabs.jsx]: done

**Pull request merged**